neptune-python-utils/neptune_python_utils/batch_utils.py

```diff
@@ -127,15 +138,15 @@ def upsert_edge(t, row, **kwargs):
         for key, value in row.items():       
             mapping = mappings.mapping_for(key)
             if not mapping.is_token():
-                create_traversal = create_traversal.property(mapping.name, mapping.convert(value))
+                create_traversal = add_property(create_traversal, mapping, value)
```

Comment:
Same comment on add_edge() - Edge properties can only have a single value.

Reply:
I've ensured that edge properties don't include a cardinality , and that lists/sets/tuples throw an error. As with bulk load CSV files, users can still supply cardinality or [] in property headings, but they'll have no impact on the insert/upsert.